test/test_image.py

```diff
@@ -0,0 +1,473 @@
+    """
+    Tests that the build method constructs and returns an Imagemage when
+    given a Dockerfile.
+    """
+    img = Image.build(tag="test")
+    inspect_process = run(
+        split("docker inspect -f='{{.Id}}' test"),
+        text=True,
+        capture_output=True)
+    id = inspect_process.stdout.strip()
+
+    assert img is not None
+    assert img.id == id
+    run(split("docker image remove test"))
+
+
+def test_build_from_dockerfile_output_to_file():
+    """
+    Tests that the build method writes to a file when configured to
+    do so.
+    """
+    file = open("testfile.txt", "w")
+    img = Image.build(
+        tag="test",
+        dockerfile="./Dockerfile",
+        stdout=file,
+        stderr=file
+    )
+    inspect_process = run(
+        split("docker inspect -f='{{.Id}}' test"),
+        text=True,
+        capture_output=True)
+    id = inspect_process.stdout.strip()
+    file.close()
+
+    assert img is not None
+    assert img.id == id
+    assert os.stat("testfile.txt").st_size > 0
+
+    os.remove("testfile.txt")
+    run(split("docker image remove test"))
+
+
+def test_build_from_dockerfile_dockerfile_in_different_location():
+    """
+    Tests that the build method can build an image from a dockerfile in a
+    different location than the context root directory.
+    """
+    img = Image.build(
+        tag="test", dockerfile="dockerfiles/alpine_functional.dockerfile"
+    )
+    inspect_process = run(
+        split("docker inspect -f='{{.Id}}' test"),
+        text=True,
+        capture_output=True)
+    id = inspect_process.stdout.strip()
+
+    assert img is not None
+    assert img.id == id
+    run(split("docker image remove test"))
+
+
+def test_build_from_dockerfile_context_in_different_location():
+    """
+    Tests that the build method can build when the context is set to a
+    different directory.
+    """
+    img = Image.build(
+        context="./dockerfiles",
+        tag="test",
+        dockerfile="./dockerfiles/alpine_functional.dockerfile"
+    )
+    inspect_process = run(
+        split("docker inspect -f='{{.Id}}' test"),
+        text=True,
+        capture_output=True)
+    id = inspect_process.stdout.strip()
+
+    assert img is not None
+    assert img.id == id
+    run(split("docker image remove test"))
+
+
+def test_build_from_dockerfile_in_malformed_location():
+    """
+    Tests that the build method raises a CalledProcessError when a malformed
+    dockerfile location is given.
+    """
+    img = None
+    with pytest.raises(CalledProcessError):
+        img = Image.build(
+            tag="test",
+            dockerfile="non_existant_directory/Dockerfile")
+    assert img is None
+
+
+def test_build_from_string():
+    """Tests that the build method builds and returns an Image when given a
+    dockerfile-formatted string."""
+    stdout: str = run(
+        split("cat Dockerfile"),
+        capture_output=True,
+        text=True).stdout
```

Comment:
```suggestion
    from pathlib import Path
    stdout: str = Path("Dockerfile").read_text()
```

https://docs.python.org/3/library/pathlib.html#pathlib.Path.read_text